src/app/modules/submission/encrypt-submission/__tests__/encrypt-submission.controller.spec.ts

```diff
@@ -88,10 +115,114 @@ describe('encrypt-submission.controller', () => {
         attachmentMetadata: mockSignedUrls,
       }
       expect(mockRes.json).toHaveBeenCalledWith(expected)
+      expect(MockUserService.getPopulatedUserById).toHaveBeenCalledWith(
+        MOCK_USER_ID,
+      )
+      expect(MockAuthService.getFormAfterPermissionChecks).toHaveBeenCalledWith(
+        {
+          user: MOCK_USER,
+          formId: MOCK_FORM_ID,
+          level: PermissionLevel.Read,
+        },
+      )
+      expect(MockEncryptSubService.checkFormIsEncryptMode).toHaveBeenCalledWith(
+        MOCK_FORM,
+      )
+    })
+
+    it('should return 400 if form is not an encrypt mode form', async () => {
+      // Arrange
+      const mockRes = expressHandler.mockResponse()
+
+      MockUserService.getPopulatedUserById.mockReturnValueOnce(
+        okAsync(MOCK_USER),
+      )
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValueOnce(
+        okAsync(MOCK_FORM),
+      )
```

Comment:
again, might be good to put this in a `beforeEach` for readability

Reply:
Resolved in 2e97f93